src/workerd/api/util.c++

```diff
@@ -52,6 +53,12 @@ kj::String toUpper(kj::String&& str) {
   return kj::mv(str);
 }
 
+bool isHexDigit(uint32_t c) {
+  // `u_isxdigit()` checks for the Latin letters a-f and A-F in both ASCII and
+  // Fullwidth ASCII. The first check (c <= 0x7f) is to ignore Fullwidth ASCII.
+  return c <= 0x7f && u_isxdigit(c);
```

Comment:
I kind of feel like at this point you might as well write out:

```
('0' <= c && c <= '9') ||
('a' <= c && c <= 'f') ||
('A' <= c && c <= 'F')
```

Using a library is "best practice", but here we found the library actually does not do what we wanted it to. Instead of adding extra checks to try to convince the library to do what we want it seems safer to just write out exactly what we want.

Plus, it's probably faster.

Reply:
I would also declare the function inline in the header -- the code it generates when inlined is probably shorter than the code to make a function call.

Reply:
Done.